includes/backup.js

```diff
@@ -43,24 +45,51 @@ module.exports = function(dbUrl, blocksize, parallelism, log, resume) {
   const start = new Date().getTime();  // backup start time
   const batchesPerDownloadSession = 50;  // max batches to read from log file for download at a time (prevent OOM)
 
-  // If resuming, pick up from existing log file from previous run. Otherwise,
-  // create new log file and process from that.
-  if (resume) {
-    downloadRemainingBatches(log, dbUrl, ee, start, batchesPerDownloadSession, parallelism);
-  } else {
-    spoolchanges(dbUrl, log, blocksize, function(err) {
-      if (err) {
-        ee.emit('error', err);
-      } else {
-        downloadRemainingBatches(log, dbUrl, ee, start, batchesPerDownloadSession, parallelism);
-      }
-    });
-  }
+  client = request.client(dbUrl, parallelism);
+
+  validateBulkGetSupport(dbUrl, function(err) {
```

Comment:
I wonder if it is worth breaking out the actual backup piece into a named function so that it is a bit easier to follow what is happening with all the callbacks.
```js
validateBulkGetSupport(dbUrl, function(err) {
  if(err) {
    return ee.emit('error', err);
  } else {
    proceedWithBackup(); // <--- contains all the if(resume), spoolchanges, download etc
  }
```